### ISSUE TITLE
Trying to install NSCP-0.4.3.65-1.x86_64.rpm on CentOS 6.5

### ISSUE THREAD

**Issue by box293** [mickem/nscp#65]
I'm not sure if you are supporting CentOS 6.5 or not so I'm posting here (FYI CentOS 7.0 installs fine with RPM).

Basically I'm getting dependency issues when try to install. Here are all my steps:

<pre>[root@centos06 ~]# cd /tmp
[root@centos06 tmp]# wget http://files.nsclient.org/beta/NSCP-0.4.3.65-1.x86_64.rpm
--2014-12-12 22:13:28--  http://files.nsclient.org/beta/NSCP-0.4.3.65-1.x86_64.rpm
Resolving files.nsclient.org... 108.162.202.46, 108.162.201.46
Connecting to files.nsclient.org|108.162.202.46|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 48934868 (47M) [application/x-redhat-package-manager]
Saving to: `NSCP-0.4.3.65-1.x86_64.rpm'

100%[========================================================================================================>] 48,934,868  1.26M/s   in 39s     

2014-12-12 22:14:07 (1.20 MB/s) - `NSCP-0.4.3.65-1.x86_64.rpm' saved [48934868/48934868]

[root@centos06 tmp]# wget http://download.fedoraproject.org/pub/epel/6/x86_64/epel-release-6-8.noarch.rpm
--2014-12-12 22:14:48--  http://download.fedoraproject.org/pub/epel/6/x86_64/epel-release-6-8.noarch.rpm
Resolving download.fedoraproject.org... 67.203.2.67, 140.211.169.197, 213.175.193.206, ...
Connecting to download.fedoraproject.org|67.203.2.67|:80... connected.
HTTP request sent, awaiting response... 302 FOUND
Location: http://epel.mirror.digitalpacific.com.au/6/x86_64/epel-release-6-8.noarch.rpm [following]
--2014-12-12 22:14:49--  http://epel.mirror.digitalpacific.com.au/6/x86_64/epel-release-6-8.noarch.rpm
Resolving epel.mirror.digitalpacific.com.au... 101.0.101.66, 2401:fc00:0:20e::2
Connecting to epel.mirror.digitalpacific.com.au|101.0.101.66|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 14540 (14K) [application/x-rpm]
Saving to: `epel-release-6-8.noarch.rpm'

100%[========================================================================================================>] 14,540      --.-K/s   in 0.04s   

2014-12-12 22:14:49 (348 KB/s) - `epel-release-6-8.noarch.rpm' saved [14540/14540]

[root@centos06 tmp]# yum -y install epel-release-6-8.noarch.rpm 
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: centos.mirror.digitalpacific.com.au
 * extras: centos.mirror.digitalpacific.com.au
 * updates: centos.mirror.digitalpacific.com.au
Setting up Install Process
Examining epel-release-6-8.noarch.rpm: epel-release-6-8.noarch
Marking epel-release-6-8.noarch.rpm to be installed
Resolving Dependencies
--> Running transaction check
---> Package epel-release.noarch 0:6-8 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

==================================================================================================================================================
 Package                            Arch                         Version                     Repository                                      Size
==================================================================================================================================================
Installing:
 epel-release                       noarch                       6-8                         /epel-release-6-8.noarch                        22 k

Transaction Summary
==================================================================================================================================================
Install       1 Package(s)

Total size: 22 k
Installed size: 22 k
Downloading Packages:
Running rpm_check_debug
Running Transaction Test
Transaction Test Succeeded
Running Transaction
  Installing : epel-release-6-8.noarch                                                                                                        1/1 
  Verifying  : epel-release-6-8.noarch                                                                                                        1/1 

Installed:
  epel-release.noarch 0:6-8                                                                                                                       

Complete!
[root@centos06 tmp]# yum -y install NSCP-0.4.3.65-1.x86_64.rpm
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
epel/metalink                                                                                                              | 3.3 kB     00:00     
 * base: mirror.rackcentral.com.au
 * epel: epel.mirror.digitalpacific.com.au
 * extras: mirror.rackcentral.com.au
 * updates: mirror.rackcentral.com.au
epel                                                                                                                       | 4.4 kB     00:00     
epel/primary_db                                                                                                            | 6.4 MB     00:05     
Setting up Install Process
Examining NSCP-0.4.3.65-1.x86_64.rpm: nscp-0.4.3.65-1.x86_64
Marking NSCP-0.4.3.65-1.x86_64.rpm to be installed
Resolving Dependencies
--> Running transaction check
---> Package nscp.x86_64 0:0.4.3.65-1 will be installed
--> Processing Dependency: boost-filesystem for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: boost-program-options for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: boost-thread for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libc.so.6(GLIBC_2.14)(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libstdc++.so.6(GLIBCXX_3.4.15)(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: protobuf for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libboost_date_time-mt.so.1.53.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libboost_filesystem-mt.so.1.53.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libboost_program_options-mt.so.1.53.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libboost_python-mt.so.1.53.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libboost_regex-mt.so.1.53.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libboost_system-mt.so.1.53.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libboost_thread-mt.so.1.53.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libcryptopp.so.6()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libprotobuf.so.8()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libpython2.7.so.1.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Running transaction check
---> Package boost-filesystem.x86_64 0:1.41.0-25.el6 will be installed
--> Processing Dependency: boost-system = 1.41.0-25.el6 for package: boost-filesystem-1.41.0-25.el6.x86_64
--> Processing Dependency: libboost_system.so.5()(64bit) for package: boost-filesystem-1.41.0-25.el6.x86_64
--> Processing Dependency: libboost_system-mt.so.5()(64bit) for package: boost-filesystem-1.41.0-25.el6.x86_64
---> Package boost-program-options.x86_64 0:1.41.0-25.el6 will be installed
---> Package boost-thread.x86_64 0:1.41.0-25.el6 will be installed
---> Package cryptopp.x86_64 0:5.6.2-2.el6 will be installed
---> Package nscp.x86_64 0:0.4.3.65-1 will be installed
--> Processing Dependency: libc.so.6(GLIBC_2.14)(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libstdc++.so.6(GLIBCXX_3.4.15)(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libboost_date_time-mt.so.1.53.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libboost_filesystem-mt.so.1.53.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libboost_program_options-mt.so.1.53.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libboost_python-mt.so.1.53.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libboost_regex-mt.so.1.53.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libboost_system-mt.so.1.53.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libboost_thread-mt.so.1.53.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libprotobuf.so.8()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libpython2.7.so.1.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
---> Package protobuf.x86_64 0:2.3.0-8.el6 will be installed
--> Running transaction check
---> Package boost-system.x86_64 0:1.41.0-25.el6 will be installed
---> Package nscp.x86_64 0:0.4.3.65-1 will be installed
--> Processing Dependency: libc.so.6(GLIBC_2.14)(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libstdc++.so.6(GLIBCXX_3.4.15)(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libboost_date_time-mt.so.1.53.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libboost_filesystem-mt.so.1.53.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libboost_program_options-mt.so.1.53.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libboost_python-mt.so.1.53.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libboost_regex-mt.so.1.53.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libboost_system-mt.so.1.53.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libboost_thread-mt.so.1.53.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libprotobuf.so.8()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Processing Dependency: libpython2.7.so.1.0()(64bit) for package: nscp-0.4.3.65-1.x86_64
--> Finished Dependency Resolution
Error: Package: nscp-0.4.3.65-1.x86_64 (/NSCP-0.4.3.65-1.x86_64)
           Requires: libboost_thread-mt.so.1.53.0()(64bit)
Error: Package: nscp-0.4.3.65-1.x86_64 (/NSCP-0.4.3.65-1.x86_64)
           Requires: libc.so.6(GLIBC_2.14)(64bit)
Error: Package: nscp-0.4.3.65-1.x86_64 (/NSCP-0.4.3.65-1.x86_64)
           Requires: libprotobuf.so.8()(64bit)
Error: Package: nscp-0.4.3.65-1.x86_64 (/NSCP-0.4.3.65-1.x86_64)
           Requires: libboost_date_time-mt.so.1.53.0()(64bit)
Error: Package: nscp-0.4.3.65-1.x86_64 (/NSCP-0.4.3.65-1.x86_64)
           Requires: libboost_filesystem-mt.so.1.53.0()(64bit)
Error: Package: nscp-0.4.3.65-1.x86_64 (/NSCP-0.4.3.65-1.x86_64)
           Requires: libboost_system-mt.so.1.53.0()(64bit)
Error: Package: nscp-0.4.3.65-1.x86_64 (/NSCP-0.4.3.65-1.x86_64)
           Requires: libboost_program_options-mt.so.1.53.0()(64bit)
Error: Package: nscp-0.4.3.65-1.x86_64 (/NSCP-0.4.3.65-1.x86_64)
           Requires: libpython2.7.so.1.0()(64bit)
Error: Package: nscp-0.4.3.65-1.x86_64 (/NSCP-0.4.3.65-1.x86_64)
           Requires: libboost_regex-mt.so.1.53.0()(64bit)
Error: Package: nscp-0.4.3.65-1.x86_64 (/NSCP-0.4.3.65-1.x86_64)
           Requires: libstdc++.so.6(GLIBCXX_3.4.15)(64bit)
Error: Package: nscp-0.4.3.65-1.x86_64 (/NSCP-0.4.3.65-1.x86_64)
           Requires: libboost_python-mt.so.1.53.0()(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest</pre>

**Comment by box293** [comment]
Not sure if I should create a new issue or not but I'll continue here.

I've tried the NSCP-0.4.3.73-1.el6.x86_64.rpm on CentOS 6.5 x64 and it installs fine, great work!

However it appears as though the nscp service is missing. There is no /etc/init.d/nscp file.

I did sort of make it work by:
<pre>wget https://github.com/mickem/nscp/raw/master/files/nscp.init -O /etc/init.d/nscp
chmod +x /etc/init.d/nscp
/etc/init.d/nscp start</pre>

This does start the service and I can perform checks against it from other systems however it does produce the following output when I start the service:

<pre>Starting nscp:
[root@centos06 init.d]# E       nrpe Failed to bind [::]:5666(ipv6): Address already in use
                    /source/nscp/include/socket/server.hpp:201
E       nrpe Failed to setup endpoint
                    /source/nscp/include/socket/server.hpp:149
Service started waiting for termination event...</pre>

**Comment by mickem** [comment]
yes you are correct... 6.5 does not have system d :)
Will fix...
As for the init script: looks like a missing & maybe? will check that as well

**Comment by mickem** [comment]
Fixed the init script in next build, please open a new ticket if there are other errors (not related to init script on 6.5), makes it simpler to tag things if there are different issues...
